src/services/compare_service.py

```
@@ -0,0 +1,223 @@
+    # 맵 생성 (예: {1: "KR", 2: "US"})
+    country_map = {c.country_id: c.country_name for c in countries}
+    # ---------------------------------------------------------
+
+    # 1. [임베딩] 질문을 벡터로 변환 (1회만 수행)
+    try:
+        text_input = TextEmbeddingInput(text=query, task_type="RETRIEVAL_QUERY")
+        embeddings = embedding_model.get_embeddings([text_input])
+        query_vector = embeddings[0].values
+    except Exception as e:
+        print(f"❌ 임베딩 실패: {e}")
+        raise e
```

Ruff (0.14.8)

Use `raise` without specifying exception name

Remove exception name

(TRY201)